hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/http/HttpServer2.java

```diff
@@ -1227,6 +1242,16 @@ public void start() throws IOException {
               .register("prometheus", "Hadoop metrics prometheus exporter",
                   prometheusMetricsSink);
         }
+        if (statsHandler != null) {
+          // Create metrics source for each HttpServer2 instance.
+          // Use port number to make the metrics source name unique.
+          int port = -1;
+          for (ServerConnector connector : listeners) {
+            port = connector.getLocalPort();
+            break;
+          }
+          metrics = HttpServer2Metrics.create(statsHandler, port);
```

Comment:
so if we have both http and https bound to the server, this metrics will cover both connectors but its name will only use one of the port?

Reply:
Yes.